src/test/java/io/jenkins/plugins/analysis/warnings/plugins/EnvInjectPluginITest.java

```diff
@@ -0,0 +1,179 @@
+        java.setPattern("**/*.txt");
+        enableWarnings(project, java);
+
+        createFileWithJavaWarnings("javac.txt", project, 1, 2);
+
+        EnvInjectBuildWrapper envInjectBuildWrapper = new EnvInjectBuildWrapper(new EnvInjectJobPropertyInfo(
+                null, "HELLO_WORLD=hello_test\nMY_ENV_VAR=42",
+                null, null, false, null
+        ));
+        project.getBuildWrappersList().add(envInjectBuildWrapper);
+
+        // Setup capturing for environment vars
+        CaptureEnvironmentBuilder capture = new CaptureEnvironmentBuilder();
+        project.getBuildersList().add(capture);
+
+        // Do the actual build
+        AnalysisResult analysisResult = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        // Assert that the injected env vars where available during the build
+        EnvVars envVars = capture.getEnvVars();
+        assertThat(envVars.get("HELLO_WORLD")).isEqualTo("hello_test");
+        assertThat(envVars.get("MY_ENV_VAR")).isEqualTo("42");
+
+        // Assert that the injectedEnvVars.txt was successfully created
+        FreeStyleBuild lastBuild = project.getLastBuild();
+        assertThat(lastBuild).isNotNull();
+        Path envFile = Paths.get(lastBuild.getRootDir().getPath(), "injectedEnvVars.txt");
+        List<String> lines = Files.readAllLines(envFile, Charset.forName("ISO-8859-1"));
+
+        assertThat(lines.contains("HELLO_WORLD=hello_test"));
+        assertThat(lines.contains("MY_ENV_VAR=42"));
+
+        assertThat(analysisResult).hasTotalSize(2);
+    }
+
+    /**
+     * Make sure that a file pattern containing environment variables correctly matches the expected files.
+     */
+    @Test
+    public void shouldResolveEnvVariablesInPattern() {
+        // Set up the project and configure the java warnings
+        FreeStyleProject project = createFreeStyleProject();
+
+        Java java = new Java();
+        java.setPattern("**/*.${FILE_EXT}");
+        enableWarnings(project, java);
+
+        // Set up the environment variable we want and one additional one
+        EnvInjectBuildWrapper envInjectBuildWrapper = new EnvInjectBuildWrapper(new EnvInjectJobPropertyInfo(
+                null, "HELLO_WORLD=hello_test\nFILE_EXT=txt",
+                null, null, false, null
+        ));
+        project.getBuildWrappersList().add(envInjectBuildWrapper);
+
+        // Create one file which matches the pattern and once which should not match
+        createFileWithJavaWarnings("javac.txt", project, 1, 2, 3);
+        createFileWithJavaWarnings("javac.csv", project, 1, 2);
+
+        AnalysisResult analysisResult = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        // Assert that the expected amount of warnings were found with the pattern
+        assertThat(analysisResult).hasTotalSize(3);
+        // Make sure that only the expected files where found
+        analysisResult.getInfoMessages().contains("-> found 1 file");
+    }
```

Comment:
Can you also check for 
```
Searching for all files in '/var/folders/pg/qr8ry2kd4qjc151jhq9ksgx00000gn/T/j h7395811023573529520/workspace/test0' that match the pattern '*.xml'
```
Or is the pattern expansion done after the logging statement?

Reply:
Seems that you resolved the conversation without giving an answer?